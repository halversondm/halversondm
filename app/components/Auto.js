/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";
import {Link} from "react-router";

const Auto = React.createClass({
  render() {
    return <div>
      <h2 className="text-primary">Autos</h2>
      <p>I've had two cars and two motorcycles that I've really enjoyed since I
        have been able to buy.</p>
      <ul>
        <li><Link to="grandPrix">2001 Grand Prix GTP</Link></li>
        <li><Link to="charger">2012 Dodge Charger R/T</Link></li>
        <li><Link to="honda">1997 Honda Shadow 750 ACE</Link></li>
        <li><Link to="yamaha">2004 Yamaha VStar 1100 Classic</Link></li>
      </ul>
    </div>;
  }
});

export default Auto;
