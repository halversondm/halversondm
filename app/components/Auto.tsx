/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import { Link } from "react-router-dom";
import { type ReactNode } from "react";

export default function Auto(): ReactNode {
  return (
    <div>
      <h2 className="text-primary">Autos</h2>
      <p>
        I&apos;ve had three cars and two motorcycles that I&apos;ve really
        enjoyed since I have been able to buy.
      </p>
      <ul>
        <li>
          <Link to="/grandPrix">2001 Grand Prix GTP</Link>
        </li>
        <li>
          <Link to="/charger">2012 Dodge Charger R/T</Link>
        </li>
        <li>
          <Link to="/honda">1997 Honda Shadow 750 ACE</Link>
        </li>
        <li>
          <Link to="/yamaha">2004 Yamaha VStar 1100 Classic</Link>
        </li>
        <li>
          <Link to="/corvette">2001 Chevrolet Corvette</Link>
        </li>
        <li>
          <Link to="/C5Secrets">C5 Corvette Tips and Secrets</Link>
        </li>
      </ul>
    </div>
  );
}
