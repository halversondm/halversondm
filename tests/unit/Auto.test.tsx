import React from "react";
import { render } from '@testing-library/react';
import Auto from "../../app/components/Auto";
import {BrowserRouter, MemoryRouter} from 'react-router-dom'


test("Auto component", () => {
    const component = render(<Auto />, {wrapper: BrowserRouter});
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});