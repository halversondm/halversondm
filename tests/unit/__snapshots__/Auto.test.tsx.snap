// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Auto component 1`] = `"<div><div><h2 class="text-primary">Autos</h2><p>I've had three cars and two motorcycles that I've really enjoyed since I have been able to buy.</p><ul><li><a href="/grandPrix">2001 Grand Prix GTP</a></li><li><a href="/charger">2012 Dodge Charger R/T</a></li><li><a href="/honda">1997 Honda Shadow 750 ACE</a></li><li><a href="/yamaha">2004 Yamaha VStar 1100 Classic</a></li><li><a href="/corvette">2001 Chevrolet Corvette</a></li><li><a href="/C5Secrets">C5 Corvette Tips and Secrets</a></li></ul></div></div>"`;
